src/SerialFirmata.h

```diff
@@ -23,6 +23,10 @@
 #include <SoftwareSerial.h>
 #endif
 
+//uncomment SERIAL_STORE_AND_FORWARD to collect bytes received by serial port until the receive buffer 
+// gets filled or a data gap is detected to avoid forwarding single bytes at baud rates below 50000
+//#define SERIAL_STORE_AND_FORWARD
```

Comment:
I think it's okay as is. I forgot about this earlier, but the better approach is to provide a way for the user to enable RX buffering through an extension to the [Serial feature protocol](https://github.com/firmata/protocol/blob/master/serial.md) at that point these ifdefs will not longer be necessary (the code will remain but will only run if a boolean is set).